lib/tools/profiler.js

```diff
@@ -0,0 +1,113 @@
+'use strict';
+
+const v8Profiler = require('v8-profiler');
+const EventEmitter = require('events');
+
+// 30 min
+const DEFAULT_INTERVAL_MS = 1800000;
+const NAME_PREFIX = 'v8.heap.snapshot';
```

Comment:
Does it make sense to conditionally check for config variables here, to avoid code changes if we want to tweak this?

```
const DEFAULT_INTERVAL_MS = process.env.PROFILER_DEFAULT_INTERVAL_MS || 1800000;
const NAME_PREFIX = process.env.PROFILER_NAME_PREFIX || 'v8.heap.snapshot';
```